webapp/src/modules/mapBuilder.js

```diff
@@ -57,60 +75,68 @@ export default class MapBuilder {
+        // prevent animation issues when zooming
+        if (this._animationTimeoutId) {
+          clearTimeout(this._animationTimeoutId);
+          this._animationTimeoutId = null;
+        }
+        this._locationAccuracyCircle._path.classList.remove(
+          "mobility-location-accuracy-circle-animation"
+        );
+        this._locationMarker._icon.style.transition = "none";
+      }
+    });
+    this._map.on("zoomend", () => {
+      if (this._locationAccuracyCircle && this._locationMarker) {
+        this._animationTimeoutId = setTimeout(() => {
+          this._locationAccuracyCircle._path.classList.add(
+            "mobility-location-accuracy-circle-animation"
+          );
+          this._locationMarker._icon.style.transition = "all 1000ms linear 0s";
+        }, 250);
+      }
     });
   }
 
-  setMarkerCluster() {
-    this._mcg = this._l.markerClusterGroup({
-      showCoverageOnHover: false,
-      maxClusterRadius: (mapZoom) => {
-        // only cluster same location markers above zoom 17
-        return mapZoom >= 17 ? 0 : 32;
-      },
-      iconCreateFunction: (cluster) => {
-        return this._l.divIcon({
-          html: "<b>" + cluster.getChildCount() + "</b>",
-          className: "mobility-map-cluster-icon",
-        });
-      },
-    });
+  setVehicleEventHandlers() {
+    this._map.on("moveend", this.moveEnd, this);
+    this._map.on("click", this.click, this);
   }
 
-  loadScooters({ onVehicleClick }) {
-    this.setEventHandlers(onVehicleClick).getScooters(onVehicleClick);
-    this._map.addLayer(this._mcg);
-    return this;
+  moveEnd() {
+    if (!this._map) {
+      return;
+    }
+    const bounds = this._map.getBounds();
+    if (
+      !this._lastBounds.contains(bounds._northEast) ||
+      !this._lastBounds.contains(bounds._southWest)
+    ) {
+      this._lastBounds = bounds;
+      this.getScooters(bounds);
+    }
   }
 
-  setEventHandlers(onVehicleClick) {
-    this._map.on("moveend", () => {
-      const bounds = this._map.getBounds();
-      if (
-        this._lastBounds.contains(bounds._northEast) === false ||
-        this._lastBounds.contains(bounds._southWest) === false
-      ) {
-        this._lastBounds = bounds;
-        this.getScooters(onVehicleClick);
-      }
-    });
+  click() {
+    if (this._vehicleClicked) {
+      this._onVehicleClick(null);
+      this._vehicleClicked = false;
+    }
+  }
 
-    this._map.on("click", () => {
-      if (this._vehicleClicked) {
-        onVehicleClick(null);
-        this._vehicleClicked = false;
-      }
-    });
+  loadScooters({ onVehicleClick }) {
+    this._onVehicleClick = onVehicleClick;
+    this.getScooters(this._lastBounds);
+    this.setVehicleEventHandlers();
+    this._map.addLayer(this._mcg);
     return this;
   }
 
-  getScooters(onVehicleClick) {
-    const { _northEast, _southWest } = this._lastBounds;
+  getScooters(bounds) {
+    const { _northEast, _southWest } = bounds;
```

Comment:
When you have state that should be coherent between calls, like the 'last bounds', please pass it in rather than set the state and continually refer to it. There are race conditions here, and it's also implicit that all of these functions rely on the state.

In general, always prefer to pass parameters in, rather than modify state. Modify state only when you need to refer to it across time- for example, `lastBounds` is needed to say, "is this value contained by the previous bounds we used." But, "request these bounds and replace scooters in them with these new scooters" is part of a single logical operation so should not refer to mutable state.